exonum/src/events/codec.rs

```diff
@@ -147,6 +163,39 @@ mod test {
         assert!(responder.decode(&mut bytes).is_err());
     }
 
+    #[test]
+    #[should_panic(expected = "Received malicious message with wrong total_len")]
+    fn decode_message_wrong_length() {
+        let data = vec![0u8, 0, 0, 0, 0, 0, 32, 0, 0, 0];
+        let mut bytes: BytesMut = BytesMut::new();
+        let (ref mut responder, ref mut initiator) = create_encrypted_codecs();
+        let raw = RawMessage::new(MessageBuffer::from_vec(data.clone()));
+        initiator.encode(raw, &mut bytes).unwrap();
+
+        responder.decode(&mut bytes).unwrap();
+    }
+
+    #[test]
+    fn decode_message_eof() {
+        let data = vec![0u8, 0, 0, 0, 0, 0, 10, 0, 0, 0];
+        let mut bytes: BytesMut = BytesMut::new();
+        let (ref mut responder, ref mut initiator) = create_encrypted_codecs();
+        let raw = RawMessage::new(MessageBuffer::from_vec(data.clone()));
+        initiator.encode(raw, &mut bytes).unwrap();
+
+        match responder.decode_eof(&mut bytes.clone()) {
+            Ok(Some(ref r)) if r == &RawMessage::new(MessageBuffer::from_vec(data.clone())) => {}
+            _ => panic!("Wrong input"),
+        };
+
+        // Emulate EOF behaviour.
+        bytes.truncate(1);
+        assert!(responder.decode_eof(&mut bytes).unwrap().is_none());
+
+        bytes.clear();
+        assert!(responder.decode_eof(&mut bytes).unwrap().is_none());
+    }
```

Comment:
Nit: It could be interesting to test a case with several messages in the buffer when the EOF is received, although it seems to be unnecessary from the code coverage perspective.

Reply:
I don't think it necessary, because `FramedRead` will be calling `decode_eof` until we reach partial message that we cannot decode.

Reply:
But I slightly modified the test and now it covers multiple messages case.